core/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -214,6 +220,69 @@ public void declareField(Parser<Value, Context> p, ParseField parseField, ValueT
         declareField((p, v, c) -> consumer.accept(v, parser.parse(p, c)), parseField, type);
     }
 
+    /**
+     * Declares a parser for fields of which names are not known upfront when creating the ObjectParser.
+     * As an example, in the aggregation output we have things like:
+     * <pre><code>
+     * "aggregations" : {
+     *   "genres" : {  <--- arbitrary aggregation name
+     *      "doc_count_error_upper_bound": 0,
+     *      "sum_other_doc_count": 0,
+     *      "buckets" : [
+     *          {
+     *              "key" : "jazz",
+     *              "doc_count" : 10
+     *              "total_number_of_ratings": { <--- arbitrary aggregation name
```

Comment:
this example is not that close to reality. we cannot really parse such an agg as we need its type in the name (type#total_number_of_ratings). That is what makes the field not unknown, which is why I am not convinced that the mechanism should be a match all.